### PR TITLE
Jobs: MCH Wildfire refactor

### LINKED ISSUE #2559: Cacbot jobs module MCH add delay to hide 'WildFire GCDs landed' after WildFire ability ends
For the Cacbot jobs module, I use the MCH to practice getting Wild Fire GCD hits during the ability's duration.
But due to the 6th shot being so close to the end of the ability's length I can't tell if I'm landing that 6th hit due to the Cacbot jobs module hiding the visual for Wild Fire GCD's the instant the ability ends.

Current design:
When Wild Fire ends, the UI element that shows how many hits you landed is hidden immediately.

Desired Design
When Wild Fire ends, the UI element that shows how many hits you landed waits about 0.5-1.0 seconds before hiding to let users know how many GCDs they land when the ability ends.

### PR DESCRIPTION
**Pull request by Echoring**
- Use Wildfire buff `count` from network log instead of counting GCD. This makes Wildfire gauge more accurate.
- Now wildfire gauge will remain displaying for 2.5s after finish Wildfire. It will change to a dimmer color at this time. ( Fixes #2559 )

Co-Authored-By: 迷子 (Maiko Tan) <19927330+MaikoTan@users.noreply.github.com>

### CODE REVIEW

File: ui/jobs/jobs.js [hunk 1]
```diff
@@ -470,6 +472,12 @@ class Bars {
     this.mobGainEffectFromYouFuncMap[effectIds] = callback;
   }
 
+  onMobLosesEffectFromYou(effectIds, callback) {
+    if (Array.isArray(effectIds))
+      effectIds.forEach((id) => this.mobLoseEffectFromYouFuncMap[id] = callback);
+    this.mobLoseEffectFromYouFuncMap[effectIds] = callback;
```
Comment (quisquous) [inline ui/jobs/jobs.js:478]: Should there be an `else` here? (Maybe also in `onYouGainEffect`?)

File: ui/jobs/components/mch.js [hunk 2]
```diff
@@ -41,105 +41,86 @@ export function setup(bars) {
...
+  const wildFireStacks = [];
+  for (let i = 0; i < 6; ++i) {
+    const d = document.createElement('div');
+    wildFireContainer.appendChild(d);
+    wildFireStacks.push(d);
+  }
+
+  let wildFireCounts = 0;
+  let wildFireActive = false;
+  const refreshWildFireGuage = () => {
```
Comment (quisquous) [inline ui/jobs/components/mch.js:85]: Guage => Gauge (typo, sorry)

File: ui/jobs/components/mch.js [hunk 3]
```diff
@@ -41,105 +41,86 @@ export function setup(bars) {
...
+        if (wildFireActive)
+          wildFireStacks[i].classList.add('active');
+        else
+          wildFireStacks[i].classList.add('fix');
+      }
+    }
+  };
+  bars.onMobGainsEffectFromYou(EffectId.Wildfire, (id, e) => {
+    wildFireActive = true;
+    wildFireCounts = e.count;
```
Comment (quisquous) [inline ui/jobs/components/mch.js:98]: Ah, this is a really nice solution!

File: ui/jobs/components/mch.js [hunk 4]
```diff
@@ -41,105 +41,86 @@ export function setup(bars) {
...
-      wildFireActive = false;
-      wildFireGCD = -1;
-      refreshWildFireGuage();
-      stacksContainer.classList.add('hide');
       wildFireBox.fg = computeBackgroundColorFrom(wildFireBox, 'mch-color-wildfire');
     }, 10000);
+    setTimeout(() => {
+      stacksContainer.classList.add('hide');
+      wildFireCounts = 0;
+    }, 12500);
```
Comment (quisquous) [inline ui/jobs/components/mch.js:123]: I think this could be longer if you wanted (5s?), but I think 2.5s is also fine if you wanted to leave it as-is.

**Review by quisquous — COMMENTED** [review]
Thanks for this!

File: ui/jobs/jobs.js (hunk 1 above)
Comment (Echoring) [reply ui/jobs/jobs.js:478]: You are right. MaikoTan said he had made a mistake here and I copied his mistake.

Comment (MaikoTan) [reply ui/jobs/jobs.js]: Sorry!ヽ(*。>Д<)o゜

**Pull request merged**